### ISSUE TITLE
[16.0] mail_tracking : JS error raised when clicking on channel

### ISSUE THREAD

**Issue by remi-filament** [OCA/social#1235]
## Module
mail_tracking

## Describe the bug
When clicking on existing channel in Discuss app (for instance general on Runboat), you get the following JS error repeated multiple times :
```
Uncaught TypeError: discuss.threadView.thread.mailbox is undefined
    compute http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:14665
    onChange http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:10552
    _executeCreatedRecordsComputes http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:10554
    _flushUpdateCycle http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:10563
    update http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:10487
    update http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:10634
    saveStateInStore http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:11298
    onFocusoutTextarea http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:11266
    mainEventHandler http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:2107
    listener http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1282
    useActiveElement http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:1496
    useEffect http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:2114
    complete http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1501
    processFiber http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:2070
    processTasks http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:2065
    frame http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:2063
    flush http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:2063
    setCounter http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1504
    _render http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1498
    render http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1495
    initiateRender http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1563
    createComponent http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:2097
    template http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js line 2057 > Function:42
    _render http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1496
    render http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1495
    initiateRender http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1563
    createComponent http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:2097
    slot2 http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js line 2057 > Function:18
    callSlot http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1665
    template http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js line 2057 > Function:8
    _render http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1496
    render http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1495
    initiateRender http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1563
    createComponent http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:2097
    slot1 http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js line 2057 > Function:13
    callSlot http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1665
    template http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js line 2057 > Function:8
    _render http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1496
    render http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1495
    initiateRender http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1563
    createComponent http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:2097
    template http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js line 2057 > Function:31
    _render http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1496
    render http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1495
    render http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1571
    batched http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1263
    notifyReactives http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1524
    set http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:1538
    add http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:392
    defaultHandler http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:662
    handleError http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:673
    start http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:679
    start http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:678
    start http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:6339
    _startServices http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:6343
    startServices http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/684-7031dd2/web.assets_backend.min.js:6336
    startWebClient http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/243-4125977/web.assets_backend_prod_only.min.js:6
    <anonymous> http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/243-4125977/web.assets_backend_prod_only.min.js:3
    def http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:9
    processJob http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:9
    processJobs http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:17
    def http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:10
    promise callback*processJobs/processJob/def< http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:10
    processJob http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:9
    processJobs http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:17
    def http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:10
    promise callback*processJobs/processJob/def< http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:10
    processJob http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:9
    processJobs http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:17
    define http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/683-b52fa3b/web.assets_common.min.js:23
    <anonymous> http://oca-social-16-0-9d46807ce4d6.runboat.odoo-community.org/web/assets/243-4125977/web.assets_backend_prod_only.min.js:15
```

## To Reproduce
**Affected versions**: 16.0

1. Connect on Runboat
2. Click on general channel

**Expected behavior**
General channel exchanges should be displayed instead of the JS error

**Additional context**
Tested with Firefox 118 and Chromium 108 (same behavior)

**Comment by remi-filament** [comment]
Hi @payen000 You may want to have a look at that issue ?

**Comment by remi-filament** [comment]
Solved with #1236
